### ISSUE TITLE
change email settings

### ISSUE THREAD

**Issue by Menz01** [pucherot/Pi.Alert#184]
i just ran the install and defaulted the email settings. figuring i could change them later. i am not finding anywhere in the GUI a place to change them. how do you change the email settings after install?

**Comment by leiweibau** [comment]
Go to your homedir and then to pialert/config/pialert.conf

**Comment by GentleHoneyLover** [comment]
How do I check if the email notifications are working? What are the notification triggers?

I think I set it up correctly but receive no notifications...

**Comment by leiweibau** [comment]
You can set a "Down Alert" for a specific device and turn the device off. After the next scan, you should get a notification.

In the next few days I will release an update in my fork that provides a function for exactly this case. It is already included in the source, but not yet in the installation package.

But because you are using Docker, you won't be able to use this feature yet ;)

**Comment by GentleHoneyLover** [comment]
Thanks! I’ll take another look!

Any plans for Shoutrrr support? That makes integration with pushiver and similar services very easy

**Comment by leiweibau** [comment]
I got Telegram up and running yesterday with Shoutrrr and will be trying out a few things over the next few days.

If and how I implement it, I don't know exactly yet, because there is still some work ahead of me.

**Comment by leiweibau** [comment]
Take a look at https://github.com/leiweibau/Pi.Alert/commit/19ffafb73ef2431b07efcbd5fd5e7e2115f54a01 or https://github.com/leiweibau/Pi.Alert/commit/d83192a4c7a8065b1da35eb607118f7bd206449a

**Comment by GroupPolicy41** [comment]
Looks like email notification is not working. Will be solve near time?

**Comment by leiweibau** [comment]
If you use "pucherot/Pi.Alert", no. The last update was over 2 years ago. For this reason, I have been maintaining my own fork for over a year now.

**Comment by GroupPolicy41** [comment]
> If you use "pucherot/Pi.Alert", no. The last update was over 2 years ago. For this reason, I have been maintaining my own fork for over a year now.

Yeah i use pucherot/Pi.Alert In this case do u have any suggestion for updated version of pi alert ? What do you mean by this (I have been maintaining my own fork for over a year now.) Another version of your own pi alert version?

**Comment by leiweibau** [comment]
If you want to have a look https://github.com/leiweibau/Pi.Alert

**Comment by GroupPolicy41** [comment]
> If you want to have a look https://github.com/leiweibau/Pi.Alert

Will check,in this case email notifications is working with this version i think am i correct?

**Comment by leiweibau** [comment]
However, be sure to make a backup of your database. A migration is possible in principle, but I can not exclude errors.

The last time I tested it, it worked. I mainly use the notification in the frontend and push services myself. But the advantage with this version is, it is taken care of when something doesn't work ;-)

**Comment by GroupPolicy41** [comment]
> However, be sure to make a backup of your database. A migration is possible in principle, but I can not exclude errors.
> 
> The last time I tested it, it worked. I mainly use the notification in the frontend and push services myself. But the advantage with this version is, it is taken care of when something doesn't work ;-)

Thanks for the information. In this case, if it's not working i'll infor you :) I'm going to install this now

**Comment by leiweibau** [comment]
> The database in this fork has been extended by some fields. To take over the database from the original Pi.Alert (pucherot), an update function is available via the "pialert-cli" in the directory ~/pialert/back. The command is then ./pialert-cli update_db

**Comment by GroupPolicy41** [comment]
> > The database in this fork has been extended by some fields. To take over the database from the original Pi.Alert (pucherot), an update function is available via the "pialert-cli" in the directory ~/pialert/back. The command is then ./pialert-cli update_db

I installed it on the new instance but looks like email notification is not working I see Test notification(s) sent but no any email, could you please check it?

**Comment by leiweibau** [comment]
For this I need more information about your eMail configuration. Which email provider do you use? Can you show me the section from pialert.conf (without password and email address)? Do you check spam folders? Can you show me the pialert.1.log?
I would take care of this tomorrow, because in my time zone it is time for me to go to bed.

**Comment by GroupPolicy41** [comment]
> For this I need more information about your eMail configuration. Which email provider do you use? Can you show me the section from pialert.conf (without password and email address)? Do you check spam folders? I would take care of this tomorrow, because in my time zone it is time for me to go to bed.

Sure, dont worry, it's not urgent. How can i access to pialert.conf file? I using debian 12 on vm. By the way I using outlook smtp config for this. This config is working on uptime kuma; you can see information below;

smtp hostname : smtp-mail.outlook.com
Port: 587

**Comment by leiweibau** [comment]
I did some more testing with my Outlook account and I get proper error messages via pialert-cli as well. I'm guessing you have 2 factor authorization enabled.
From the looks of it, it seems to be out of my control though. I will do some more research later, however I have little hope of being able to offer a solution here.

```
smtplib.SMTPAuthenticationError: (535, b'5.7.3 Authentication unsuccessful
```
https://techcommunity.microsoft.com/t5/security-compliance-and-identity/smtp-xoauth-suddenly-failing-for-outlook-personal-with-535-5-7-3/m-p/3738000

**Comment by GroupPolicy41** [comment]
> I did some more testing with my Outlook account and I get proper error messages via pialert-cli as well. I'm guessing you have 2 factor authorization enabled. From the looks of it, it seems to be out of my control though. I will do some more research later, however I have little hope of being able to offer a solution here.
> 
> ```
> smtplib.SMTPAuthenticationError: (535, b'5.7.3 Authentication unsuccessful
> ```
> 
> https://techcommunity.microsoft.com/t5/security-compliance-and-identity/smtp-xoauth-suddenly-failing-for-outlook-personal-with-535-5-7-3/m-p/3738000

No, i did not enabled 2 factor authorization on the account. This account created just for smtp emails without any security options (like two factor aut)

**Comment by leiweibau** [comment]
I clicked again a new outlook address and have not activated 2FA. I was again with the error

```
smtplib.SMTPAuthenticationError: (535, b'5.7.3 Authentication unsuccessful ...
```
confronted. When I tried to login again via Outlook.com, my address didn't seem to exist anymore and I had to register again. Suddenly the email notification worked. At least I got a different error message at the beginning ;-)

```
smtplib.SMTPDataError: (554, b'5.2.252 SendAsDenied; ...
```

However, this was due to my configuration:

```
REPORT_FROM = 'Pi.Alert <your_mai@outlook.com>'
```
or

```
REPORT_FROM = 'your_mai@outlook.com'
```

Here must be your email address. After that the notification worked. I can't tell you where the error was before. Only so much that I have not made any code change 🤷‍♂️

**Comment by GroupPolicy41** [comment]
Here must be your email address. After that the notification worked. I can't tell you where the error was before. Only so much that I have not made any code change.

Good news :) Which smtp config do you use? If is it possible can you share screen for the config?

**Comment by leiweibau** [comment]
Of course. I have made some elements unrecognizable.

One thing in the beginning for explanation:
The option `REPORT_FROM = 'Pi.Alert <my_mail@outlook.com>'` differs from the syntax, because my regular mail provider has no email address as username and so the original syntax is not possible.

```
# Mail-Account Settings
# ----------------------
SMTP_SERVER       = 'smtp-mail.outlook.com'
SMTP_PORT         = 587
SMTP_USER         = 'my_mail@outlook.com'
SMTP_PASS         = 'secure_random_password'
SMTP_SKIP_TLS	  = False
SMTP_SKIP_LOGIN	  = False
```

```
# Mail Reporting
# ----------------------
REPORT_MAIL          = True
REPORT_MAIL_WEBMON   = True
REPORT_FROM          = 'Pi.Alert <my_mail@outlook.com>'
...
```

Testmail Header:
```
Return-path: <my_mail@outlook.com>
Original-recipient: rfc822;receiving_address@mail.com
Received: from m1p00im-qufv1601.mail.com by p110-mailgateway-smtp-66dffbb44c-hwvwb (mailgateway 2324B162)
	with SMTP id 55524bad-6ea2-4ef5-8a36-6e4659778bed 
	for <receiving_address@mail.com>; Mon, 11 Sep 2023 17:47:23 GMT
X-Apple-MoveToFolder: INBOX 
X-Apple-Action: MOVE_TO_FOLDER/INBOX
X-Apple-UUID: 55524bad-6ea2-4ef5-8a36-6e4659778bed
Received: from GBR01-LO2-obe.outbound.protection.outlook.com (mail-lo2gbr01olkn2090.outbound.protection.outlook.com [40.92.78.90])
	by m1p00im-qufv1601.mail.com (Postfix) with ESMTPS id 7C9664045BC
	for <receiving_address@mail.com>; Mon, 11 Sep 2023 17:47:20 +0000 (UTC)
X-ICL-SCORE: 3.333033030043
...
```

**Comment by GroupPolicy41** [comment]
> Of course. I have made some elements unrecognizable.
> 
> One thing in the beginning for explanation: The option `REPORT_FROM = 'Pi.Alert <my_mail@outlook.com>'` differs from the syntax, because my regular mail provider has no email address as username and so the original syntax is not possible.
> 
> ```
> # Mail-Account Settings
> # ----------------------
> SMTP_SERVER       = 'smtp-mail.outlook.com'
> SMTP_PORT         = 587
> SMTP_USER         = 'my_mail@outlook.com'
> SMTP_PASS         = 'secure_random_password'
> SMTP_SKIP_TLS	  = False
> SMTP_SKIP_LOGIN	  = False
> ```
> 
> ```
> # Mail Reporting
> # ----------------------
> REPORT_MAIL          = True
> REPORT_MAIL_WEBMON   = True
> REPORT_FROM          = 'Pi.Alert <my_mail@outlook.com>'
> ...
> ```
> 
> Testmail Header:
> 
> ```
> Return-path: <my_mail@outlook.com>
> Original-recipient: rfc822;receiving_address@mail.com
> Received: from m1p00im-qufv1601.mail.com by p110-mailgateway-smtp-66dffbb44c-hwvwb (mailgateway 2324B162)
> 	with SMTP id 55524bad-6ea2-4ef5-8a36-6e4659778bed 
> 	for <receiving_address@mail.com>; Mon, 11 Sep 2023 17:47:23 GMT
> X-Apple-MoveToFolder: INBOX 
> X-Apple-Action: MOVE_TO_FOLDER/INBOX
> X-Apple-UUID: 55524bad-6ea2-4ef5-8a36-6e4659778bed
> Received: from GBR01-LO2-obe.outbound.protection.outlook.com (mail-lo2gbr01olkn2090.outbound.protection.outlook.com [40.92.78.90])
> 	by m1p00im-qufv1601.mail.com (Postfix) with ESMTPS id 7C9664045BC
> 	for <receiving_address@mail.com>; Mon, 11 Sep 2023 17:47:20 +0000 (UTC)
> X-ICL-SCORE: 3.333033030043
> ...
> ```

I think now it's okay. i Sharing with you old config; Looks like the problem was " **Report From"**

# Mail-Account Settings
# ----------------------
SMTP_SERVER       = 'smtp-mail.outlook.com'
SMTP_PORT         = 587
SMTP_USER         = 'xyz@outlook.com'
SMTP_PASS         = 'xyzpassword'
SMTP_SKIP_TLS	  = False
SMTP_SKIP_LOGIN	  = False

# WebGUI Reporting
# ----------------------
REPORT_WEBGUI        = True
REPORT_WEBGUI_WEBMON = True

# Mail Reporting
# ----------------------
REPORT_MAIL          = True
REPORT_MAIL_WEBMON   = True
REPORT_FROM          = 'Pi.Alert < + SMTP_USER +>'
REPORT_TO            = 'xyz@outlook.com'
REPORT_DEVICE_URL    = 'http://pi.alert/deviceDetails.php?mac='
REPORT_DASHBOARD_URL = 'http://pi.alert/'

I changed it with Email account and i could sent test email. I'm not sure why it seems like this, normally should be added email account in there because during installation i already wrote.

**Comment by leiweibau** [comment]
I am happy that I was able to help.

The setup is still from the main project and I don't see myself changing it at the moment, because I don't know everything. I am already satisfied that I can adapt Pi.Alert to my wishes and ideas of others ;-)

**Comment by GroupPolicy41** [comment]
> I am happy that I was able to help.
> 
> The setup is still from the main project and I don't see myself changing it at the moment, because I don't know everything. I am already satisfied that I can adapt Pi.Alert to my wishes and ideas of others ;-)

You've been really helpful. Thanks a lot. I also appreciate that you responded to me on this issue despite the time difference. I just want to ask one last question. Is there a way to update the time in the top right corner on the home page? I've already changed the time zone for my Debian server and reports now show according to the local timezone but that didn't fix the time on the homepage. Not in the local time zone.

**Comment by leiweibau** [comment]
When you look in the help/FAQ section, open the 1. question box. The "red marked" php.ini path, leads you to the file, you need to change.

> The clock on the top right and the times of the events/presence are not correct (time difference).

**Comment by GroupPolicy41** [comment]
Ah okay, i saw. Thanks a lot again for all!

**Comment by tismofied** [comment]
Is gmail smtp working  without having to set up an  app password? When I used my regular password I didnt get any emails

**Comment by leiweibau** [comment]
If you have 2 factor authentication on your google account, you will need a separate app password.
